src/cargo/ops/cargo_rustc/fingerprint.rs

```diff
@@ -82,8 +82,11 @@ pub fn prepare_target<'a, 'cfg>(cx: &mut Context<'a, 'cfg>,
         missing_outputs = !root.join(unit.target.crate_name())
                                .join("index.html").exists();
     } else {
-        for (filename, _) in try!(cx.target_filenames(unit)) {
-            missing_outputs |= fs::metadata(root.join(filename)).is_err();
+        for (src, link_dst, _) in try!(cx.target_filenames(unit)) {
+            missing_outputs |= fs::metadata(&src).is_err();
```

Comment:
This was actually kept around from literally Rust 1.0.0, but nowadays it's fine to change this to `!src.exists()`